build/build.go

```diff
@@ -591,24 +592,60 @@ func (s *Session) buildImportPathWithSrcDir(path string, srcDir string) (*Packag
 	return pkg, archive, nil
 }
 
+const (
+	hashDebug = false
+)
+
 func (s *Session) BuildPackage(pkg *PackageData) (*compiler.Archive, error) {
 	if archive, ok := s.Archives[pkg.ImportPath]; ok {
 		return archive, nil
 	}
 
+	// For non-main and test packages we build up a hash that will help
+	// determine staleness. Set hashDebug to see this in action. The format is:
+	//
+	// ## sync
+	// gopherjs bin: 0x519d22c6ab65a950f5b6278e4d65cb75dbd3a7eb1cf16e976a40b9f1febc0446
```

Comment:
Hmm, there could be a better name but I don't have a good idea. At least, I feel like the keys should end with `hash` for consistency?

Reply:
Ok I've made it `gopherjs binary hash` to be explicit. I've also moved the calculation of this hash into a separate function because this need only be calculated once.